stock_receipt_lot_info/tests/test_receipt_lot_info.py

```diff
@@ -0,0 +1,108 @@
+# Copyright 2022 ForgeFlow, S.L.
+# License LGPL-3.0 or later (https://www.gnu.org/licenses/lgpl.html).
+
+from datetime import timedelta
+
+from odoo.fields import Datetime
+from odoo.tests.common import TransactionCase
+
+
+class ReceiptLotInfo(TransactionCase):
+    @classmethod
+    def setUpClass(cls):
+        super(ReceiptLotInfo, cls).setUpClass()
+        group_stock_multi_locations = cls.env.ref("stock.group_stock_multi_locations")
+        cls.env.user.write({"groups_id": [(4, group_stock_multi_locations.id, 0)]})
+        cls.stock_location = cls.env.ref("stock.stock_location_stock")
+        cls.customer_location = cls.env.ref("stock.stock_location_customers")
+        cls.supplier_location = cls.env.ref("stock.stock_location_suppliers")
+        cls.uom_unit = cls.env.ref("uom.product_uom_unit")
+        cls.uom_dozen = cls.env.ref("uom.product_uom_dozen")
+        cls.product = cls.env["product.product"].create(
+            {
+                "name": "Product A",
+                "type": "product",
+                "categ_id": cls.env.ref("product.product_category_all").id,
+            }
+        )
+        cls.product_serial = cls.env["product.product"].create(
+            {
+                "name": "Product B",
+                "type": "product",
+                "tracking": "serial",
+                "categ_id": cls.env.ref("product.product_category_all").id,
+            }
+        )
+        cls.product_lot = cls.env["product.product"].create(
+            {
+                "name": "Product C",
+                "type": "product",
+                "tracking": "lot",
+                "categ_id": cls.env.ref("product.product_category_all").id,
+            }
+        )
+
+    def test_in_1(self):
+        """This tests adds info about dates in move lines and check if
+        its set correctly.
+        """
+
+        picking = self.env["stock.picking"].create(
+            {
+                "location_id": self.supplier_location.id,
+                "location_dest_id": self.stock_location.id,
+                "picking_type_id": self.env.ref("stock.picking_type_in").id,
+            }
+        )
+        move1 = self.env["stock.move"].create(
+            {
+                "name": "test_lot_info_1",
+                "location_id": self.supplier_location.id,
+                "location_dest_id": self.stock_location.id,
+                "product_id": self.product_lot.id,
+                "product_uom": self.uom_unit.id,
+                "product_uom_qty": 1.0,
+                "picking_id": picking.id,
+                "picking_type_id": self.env.ref("stock.picking_type_in").id,
+            }
+        )
+        move1._action_confirm()
+        move1._action_assign()
+        self.assertEqual(move1.state, "assigned")
```

Comment:
I would do this at picking level because that is what a user would do, users confirm and assign pickings with the specific button in the picking form view, they don't do that at move level.

Reply:
Done!